src/main/java/org/assertj/core/error/AnyElementShouldMatch.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2017 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import org.assertj.core.presentation.PredicateDescription;
+
+public class AnyElementShouldMatch extends BasicErrorMessageFactory {
+  private static final String SINGLE_NON_MATCHING_ELEMENT = "%nExpecting any element of:%n  <%s>%nto match %s predicate but this element did not:%n  <%s>";
```

Comment:
the error message is logically incorrect, it should be:
`"%nExpecting any element of:%n  <%s>%nto match %s predicate but non did."`